### DISCUSSION TITLE
Android 12 update - lost signal, consistent message to pair

### DISCUSSION THREAD

**Discussion by corn49** [NightscoutFoundation/xDrip#2064] [Q&A]
Just the other day, I ran the most recent Android 12 update. Since then, Xdrip hasn't been able to work right. I tried several steps with Navid on the FB group, but we were unable to come to resolve the issue. 

I'm on a Google Pixel 6XL, using the most recent Xdrip Nightly from April 3rd and have made sure I have settings running according to documentation. I have tried it with OB1 unbinding off to no avail. At most, I can get 3 readings in a row. 

Not sure what other information I can provide, but I would love any support I can get. This software has enabled me to live a full life in many respects and I am so grateful for the work it takes to maintain it. 

I did share my settings via imgur earlier and here they are: https://imgur.com/a/ceQYHDN

For clarity, running a G6.

**Comment by Navid200** [comment]
@jamorham Pixel 6 comes up on facebook for intermittent connectivity issues.
Some Pixel 6 users have reported success after having set Bluetooth battery usage to unrestricted.
But, in this case, that didn't help either.

**Reply by CLKRUN** [reply]
I also set unrestricted but did not work.

Pixel 6 
A12
xDrip build 2022.04.22

So it's a Pixel 6 , A12 or xDrip+ Problem?

**Comment by felix-git-hub** [comment]
> 

did you  try : setting - less common setting - bluetooth setting - Use scanning(scan before connecting on xBridge and libre bluetooth)

**Reply by corn49** [reply]
Does it matter if I am using G6 and not libre?

**Reply by Navid200** [reply]
Yes, it does.  
Android disabling apps running in the background affects Libre and Dexcom sensors.  But, how we use xDrip and how we set it up to avoid such problems could be different depending on which sensor you are using.

Please don't enable any Libre-related settings, unless you are told by the person I have tagged above.  And that will be something new for me to.  If that happens, I will need to update instructions.

**Reply by corn49** [reply]
Got it. So do not try enabling that setting.

**Reply by corn49** [reply]
@Navid200, think the app mentioned here could help?

https://github.com/Navid200/xDrip/discussions/171#discussion-3769089

**Reply by Navid200** [reply]
You can try it.  None of us has Pixel 6.
Lots of times, someone tries something and it happens that at the same time, the problem is resolved for some other reason.  They report it, as they should.  But, that doesn't necessarily make it a solution.

Unless enough people report identical results, there is no way anyone can claim they have found a solution.

**Comment by ChanceHarrison** [comment]
@corn49 I have a Pixel 6 (non Pro/XL) that I use with my G6. At first, I was having a similar issue (frequent pairing requests, intermittent connectivity, etc.). I seem to have landed on settings that have (mostly *) remedied my issues, and they are quite similar to the ones you shared. I haven't seen a pairing request in what feels like forever. 

The only difference between my settings and your settings that I can tell is that I have "Minimize scanning" unchecked while you have it checked. It might be worth seeing if unchecking that helps? I haven't tried any other workarounds of note that I can recall (beyond going through [these recommended settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html), with disabling "Allow OB1 unbonding" and disabling scanning optimizations being notable exceptions).

* My intermittent connectivity issues haven't gone away entirely, but they're basically a non-issue at this point. I have found that sometimes xDrip still will need 5-15 minutes to get out of an error state, but readings are backfilled and I barely notice any issues. RealtimeCap is consistently around 96%, at least over the last 14 days.

**Comment by speedydaniel** [comment]
Hello all,

Unfortunately, I have the same problem. After the update to Android 12 (OnePlus 8T), xDrip+ only reads out the data every 10-15 minutes. This is quite annoying for me, as I use the app for biking and have the values announced every 5 minutes.

The problem persists even after a factory reset.

On another device (OnePlus 6T) with A11, it works perfectly. 
I will now switch back to A11.

Has anyone had any other experiences with A12 and xDrip+?

**Reply by felix-git-hub** [reply]
what does the log imply? Are you using libre sensors?

**Reply by Navid200** [reply]
No, this thread is about Dexcom G6.

**Comment by jbr7rr** [comment]
Hey. Using a pixel 6 pro. I'm currently running a dexcom one in a xDrip variant in adition to a libre2 on the main app. I noticed the same issue. I reduced it somewhat by fellowing the steps on the support page by @Navid200 but it's still there. 

I have narrowed it down in the logs to the auth part of the state machine:

Happy Flow:
![Screenshot_20220509-070004](https://user-images.githubusercontent.com/60566713/167344013-b5e910d9-f60c-4a28-b540-d11fe1b3e39e.png)

Error flow:
![markup_3236](https://user-images.githubusercontent.com/60566713/167344042-31aec994-8d27-419b-848e-3b77fc09b1fd.png)

If anyone has some ideas that will be great. Will be looking further in the code myself as soon as I have some spare time


Edit: here is another error flow, now it says it is still bonded (this is what I see most of the time):
![Screenshot_20220509-183218](https://user-images.githubusercontent.com/60566713/167456022-65c0bc32-92b1-475c-be67-7d585b6955f7.png)

**Comment by Navid200** [comment]
@jamorham

**Comment by ddellspe** [comment]
On Pixel 6 Pro
G6
Recommended settings except "Allow OB1 unbonding"
xDrip+ as primary bluetooth connection, Tandem t:slim X2 as secondary connection
xDrip+ version latest (2022.05.19) but trying older ones where I don't remember having problems to try to trace things back

I'm facing similar issues, I've uploaded logs, but I'm remembering that I was previously having issues around the start of may after updating, and I've been trying to update more regularly.  I'm currently on the 2022.04.26 build and things are a bit more stable than they are with the latest nightly build (2022.05.19).  Seeing the same errors as others "Device is marked as bonded but we are being asked to bond so attempting to unbond first" followed by "Would have tried to unpair but preference setting prevents it. (unbond)".  I have the "allow unbonding" unchecked because with it, I get way too many pair requests (about every 10 minutes or so).

**Reply by jbr7rr** [reply]
Have you seen https://github.com/NightscoutFoundation/xDrip/pull/2121 ? Are you willing to try it to see if that fixes it?

**Reply by ddellspe** [reply]
I can give that a try, let me see if I can build it locally and let you know.

**Reply by ddellspe** [reply]
This has worked consistently for me for the last 3 days

**Comment by Navid200** [comment]
Please try the latest Nightly:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2022.06.09

It has changes meant to help with maintaining connectivity.

**Reply by ddellspe** [reply]
Is there some setting we need to update to use this, or is it just built in?

**Reply by Navid200** [reply]
It's built in

**Reply by jbr7rr** [reply]
Pixel 6 pro here. can confirm workaround is working in the latest Nightly

**Reply by ChanceHarrison** [reply]
Pixel 6 here. Allow OB1 Bonding enabled (used to be disabled prior to workaround). The workaround has shown up in the logs 10 times since updating, and my real-time capture is very high (97% with the few missed readings due to being out-of-range or blocking the signal). I'm cautiously optimistic that the workaround works!

**Reply by ghost** [reply]
> Pixel 6 pro here. can confirm workaround is working in the latest Nightly

I would like to buy a Pixel 6 pro and use it with xdrip+ and libre 2. I red many issue related to the bluethoot and Android 13 with pixel 6 pro. Any advice ?

**Reply by spanosp** [reply]
im using a pixel 6 pro with libre 2 and miaomiao with near no connectivity issues

**Reply by nigelkeirross** [reply]
@spanosp I would very much like to know how you are connecting to Libre2, xDrip+ & miaomiao. I currently use Libre1 with mm2 & xDrip+ (standard from Aufust 2022), and need to move to Libre2 here in UK, but am confused about options. Do you also use OOP2 app on your phone? What is the firmware version of your mm? I am running on Pixel5 with Android 13.

**Reply by spanosp** [reply]
@nigelkeirross I currently use Libre 1 (sorry, not Libre 2) with xDrip+ and miaomiao 3. I initiate the sensor with the LibreLink app and then use xDrip+. I'm running on Pixel 6 pro with Android 13 and as per xdrip the miaomiao3/Tomato firmware is 0003. I dont have many connectivity issues - I've only noticed that the phone wont automatically reconnect to the miaomiao3 after disconnecting when falling out of BT range. Not sure if this helps?

**Comment by nigelkeirross** [comment]
Thanks for the reply. I had thought you were using Libre2.
With my current setup (pixel5, Android 13, xDrip from 2022-08-03,I also
suffer strange behaviour when going out of BT range, but generally does
reconnect. With both bubble and MM2, if I go out of range, even for a as
short time (30 seconds, say), then return, it will generally reconnect, BUT
the 5 minute clock starts again and I get my next reading 5 minute after
the reconnection, thus disturbing the 5 minute cycle. So if I walk away
from my phone for 5 seconds after 4+ mins since last reading, it will be ~9
mins between readings ..

**Reply by Navid200** [reply]
@nigelkeirross Please don't use email to post in discussions or issues in this repository.
It makes it hard for everyone to follow the thread.  Please edit your post and delete the signature and quoted message included by your email client.
